src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -12487,3 +12551,107 @@ pbsv1mod_meth_release_nodes(PyObject *self, PyObject *args, PyObject *kwds)
 	hook_set_mode = hook_set_mode_orig;
 	return (py_return);
 }
+
+
+/**
+ *
+ * @brief
+ *	Returns a Python List of _server_attribute objects or  or NULL on error.
+ * @param[in]	phead	- pointer to the head of the list containing data.
+ *
+ * @return 	PyObject *
+ * @retval	<object>	- the Python list object holding
+ * @retval				the _server_attribute objects.
+ * @retval	NULL		- if an error occurred.
+ */
+PyObject *svrattrl_list_to_pyobject(pbs_list_head *phead)
+{
+	svrattrl *plist = NULL;
+	PyObject* py_list = PyList_New(0);
+
+	if (phead == NULL) {
+		log_err(errno, __func__, "NULL input parameters!");
+		Py_CLEAR(py_list);
+		return NULL;
+	}
+
+	for (plist = (svrattrl *)GET_NEXT(*phead); plist != NULL;
+		plist = (svrattrl *)GET_NEXT(plist->al_link)) {
+		PyObject *py_server_attribute = svrattrl_to_server_attribute(plist);
+		if (py_server_attribute) {
+			svrattrl *slist = NULL;
+			PyObject* py_slist = PyObject_GetAttrString(py_server_attribute, "sisters");
+			if (py_slist) {
+				for(slist = plist->al_sister; slist != NULL; slist = slist->al_sister) {
+					PyObject *py_server_attribute_sister = svrattrl_to_server_attribute(slist);
+					if (py_server_attribute_sister) {
+						PyList_Append(py_slist, py_server_attribute_sister);
+						Py_CLEAR(py_server_attribute_sister);
+					} else {
+						snprintf(log_buffer, LOG_BUF_SIZE-1,
+							"could not translate the sister for attribute <%s>", plist->al_name);
+						log_buffer[LOG_BUF_SIZE-1] = '\0';
+						log_err(PBSE_INTERNAL, __func__, log_buffer);
+						break;
+					}
+				}
+			}/* else {
+				log_err(PBSE_INTERNAL, __func__,
+					"failed to acquire sisters in server_attribute object");
+			} */
+			PyList_Append(py_list, py_server_attribute);
+			Py_CLEAR(py_server_attribute);
+		}
+	}
+	return py_list;
+}
+
+/**
+ *
+ * @brief
+ *	Returns a Python _server_attribute object or NULL on error.
+ * @param[in]	attribute	- pointer to the head of the list containing data.
+ *
+ * @return 	PyObject *
+ * @retval	<object>	- the Python _server_attribute object.
+ * @retval	NULL		- if an error occurred.
+ */
```

Comment:
Same here too, please add a @note that says the returned \<object> contains a new reference that must be Py_CLEAR-ed later.

Reply:
Note was added.